modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/restapi/ServiceCatalogRestAPITestCase.java

```diff
@@ -0,0 +1,511 @@
+        try {
+            restAPIServiceCatalog.importService(servicesFileTwo, false, null);
+        } catch (ApiException e) {
+            Assert.assertEquals(HttpStatus.SC_BAD_REQUEST, e.getCode());
+        }
+
+        // Import service with existing name and version with overwrite = true
+        ServiceInfoListDTO importServiceOverwriteRes = restAPIServiceCatalog.importService(servicesFileTwo, true, null);
+        Assert.assertNotNull(importServiceOverwriteRes);
+        Assert.assertNotNull(importServiceOverwriteRes.getList());
+        Assert.assertEquals(importServiceOverwriteRes.getList().get(0).getName(), "Pizzashack-Endpoint");
+        Assert.assertEquals(importServiceOverwriteRes.getList().get(0).getKey(), "Pizzashack-Endpoint-1.0.0");
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Export Service through the Service Catalog Rest API",
+            dependsOnMethods = "testImportService")
+    public void testExportService() throws Exception {
+
+        // Export Services
+        File exportServiceRes = restAPIServiceCatalog.exportService("Pizzashack-Endpoint", "v1");
+        Assert.assertNotNull(exportServiceRes);
+
+        // Export services with wrong name or version
+        try {
+            restAPIServiceCatalog.exportService("Pizzashack", "v1");
+        } catch (ApiException e) {
+            Assert.assertEquals(HttpStatus.SC_NOT_FOUND, e.getCode());
+        }
+    }
+
+    @Test(groups = {"wso2.am"}, description = "Create an API Through the Publisher Rest API",
+            dependsOnMethods = "testExportService")
+    public void testCreateAnAPIThroughPublisher() throws Exception {
+
+        APIDTO apiCreationDTO = new APIDTO();
+        apiCreationDTO.setName("PizzaShackAPI");
+        apiCreationDTO.setDescription("This is a simple API for Pizza Shack online pizza delivery store.");
+        apiCreationDTO.setContext("pizza");
+        apiCreationDTO.setVersion("1.0.0");
+        apiCreationDTO.setProvider("admin");
+        apiCreationDTO.setLifeCycleStatus("CREATED");
+
+        apiCreationDTO.setType(APIDTO.TypeEnum.HTTP);
+        apiCreationDTO.setAudience(APIDTO.AudienceEnum.PUBLIC);
+        apiCreationDTO.setIsDefaultVersion(false);
+
+        apiCreationDTO.setAccessControl(APIDTO.AccessControlEnum.NONE);
+
+        APIBusinessInformationDTO apiBusinessInformationDTO = new APIBusinessInformationDTO();
+        apiBusinessInformationDTO.setBusinessOwner("businessowner");
+        apiBusinessInformationDTO.setBusinessOwnerEmail("businessowner@wso2.com");
+        apiBusinessInformationDTO.setTechnicalOwner("technicalowner");
+        apiBusinessInformationDTO.setTechnicalOwnerEmail("technicalowner@wso2.com");
+        apiCreationDTO.setBusinessInformation(apiBusinessInformationDTO);
+
+        APIOperationsDTO apiOperationsDTOSampleOne = new APIOperationsDTO();
+        apiOperationsDTOSampleOne.setVerb("GET");
+        apiOperationsDTOSampleOne.setTarget("/menu");
```

Comment:
Do we need to set this parameters in API creation request? Ideally, the swagger definition should be picked from service configuration